src/ik.rs

```diff
@@ -275,18 +291,34 @@ where
         constraints: &Constraints,
     ) -> Result<(), Error> {
         let operational_space = define_operational_space(&constraints);
+        let required_dof = operational_space.iter().filter(|x| **x).count();
         let orig_positions = arm.joint_positions();
-        let use_dof = operational_space.iter().filter(|x| **x).count();
-        if orig_positions.len() < use_dof {
+        let available_dof = arm.dof() - constraints.ignored_joint_names.len();
+        if available_dof < required_dof {
             return Err(Error::PreconditionError {
-                dof: orig_positions.len(),
-                necessary_dof: use_dof,
+                dof: available_dof,
+                necessary_dof: required_dof,
             });
         }
+        let mut ignored_joint_indices = Vec::new();
+        for joint_name in constraints.ignored_joint_names.iter() {
```

Comment:
We can use this if you want.

```suggestion
        for joint_name in &constraints.ignored_joint_names {
```